runner/src/main.rs

```diff
@@ -290,64 +291,107 @@ fn build_server(opt: &BuildServer) -> Step {
+                    }
+                } else {
+                    hashmap! {}
+                },)
+            ],
         ].into_iter()
             .flatten()
             .collect::<Vec<_>>()
     }
 }
 
+fn build_functions_server(opt: &BuildFunctionsServer) -> Step {
+    Step::Multiple {
+        name: "server".to_string(),
+        steps: vec![
+            vec![Step::Single {
+                name: "create bin folder".to_string(),
+                command: Cmd::new(
+                    "mkdir",
+                    vec!["-p".to_string(), "oak_functions/loader/bin".to_string()],
+                ),
+            }],
+            vec![match opt.server_variant {
+                FunctionsServerVariant::Unsafe => build_server_step(
+                    "oak_functions/loader",
+                    &opt.server_rust_toolchain,
+                    opt.server_rust_target
+                        .as_deref()
+                        .unwrap_or(DEFAULT_SERVER_RUST_TARGET),
+                    "oak-unsafe",
+                    &hashmap! {},
+                ),
+                FunctionsServerVariant::Base => build_server_step(
+                    "oak_functions/loader",
+                    &opt.server_rust_toolchain,
+                    opt.server_rust_target
+                        .as_deref()
+                        .unwrap_or(DEFAULT_SERVER_RUST_TARGET),
+                    "",
+                    &hashmap! {},
+                ),
+            }],
+        ]
+        .into_iter()
+        .flatten()
+        .collect::<Vec<_>>(),
+    }
+}
+
+fn build_server_step(
+    manifest_dir: &str,
+    server_rust_toolchain: &Option<String>,
+    server_rust_target: &str,
+    features: &str,
+    env: &HashMap<String, String>,
+) -> Step {
+    Step::Single {
+        name: "build server".to_string(),
+        command: Cmd::new_with_env(
+            "cargo",
+            spread![
+                ...match server_rust_toolchain {
+                    // This overrides the toolchain used by `rustup` to invoke the actual
+                    // `cargo` binary.
+                    // See https://github.com/rust-lang/rustup#toolchain-override-shorthand
+                    Some(server_rust_toolchain) => vec![format!("+{}", server_rust_toolchain)],
+                    None => vec![],
+                },
+                "build".to_string(),
+                format!("--manifest-path={}/Cargo.toml", manifest_dir),
+                format!("--out-dir={}/bin", manifest_dir),
+                // `--out-dir` is unstable and requires `-Zunstable-options`.
+                "-Zunstable-options".to_string(),
+                ...if !features.is_empty() {
+                    vec![format!("--features={}", features),
+                        format!("--target={}", server_rust_target), "--release".to_string(),
```

Comment:
Why is this repeated twice in both branches of this if? It seems that --target and --release are always there, so I don't see why they need to be conditional on anything.

Reply:
Fixed.